saturation.go

```diff
@@ -0,0 +1,148 @@
+package raft
+
+import (
+	"math"
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+// saturationMetric measures the saturation (percentage of time spent working vs
+// waiting for work) of an event processing loop, such as runFSM. It reports the
+// saturation as a gauge metric (at most) once every reportInterval.
```

Comment:
Oh sure! What I was trying to get at here is that it's possible that the goroutine is asleep or busy for a period longer than `reportInterval` so we actually update the gauge less frequently. It shouldn't really be a problem as far as the metric's accuracy goes, because it's unlikely you'd drop from high saturation to dormant suddenly or vice versa.

Maybe this would be clearer:

> It attempts to report the saturation as a gauge metric once every `reportInterval`, but it may happen less frequently if the event loop is sleeping or busy for a long period.

Reply:
Ah that makes sense!